Attribute content area renderer

wezz opened the pull request:
This is the actual attribute content area renderer. 
It supports Layout Blocks that can have properties annotated with BlockRenderingMetadataAttribute's. 
That way you can expand layout blocks with additional metadata that will be rendered as attributes without having to affect the original BlockRenderingMetaData model that is included in this project. 

Ex. of such an implementation:
<img width="791" alt="image" src="https://user-images.githubusercontent.com/1185533/189166496-62c38c7b-6b37-48cf-9a93-ce16a3d1473f.png">

Commit:
Hide block from being visible to editors. (Can be overriden in admin or inherited)

Commit:
Attribute Content Area render that makes use of block metadata

Commit:
Ensure the property attribute dictates the custom attributes

vansterhant reviewed (commented):
I really think we should try to not have two nearly Identical Renderers, especially since what's in RenderContententAreaItems basically is the entire algorithm for the module.

I know you have a lot on your plate @wezz. If it's ok with you, I could take a stab at deduping it somehow?

LinusEkstrom commented:
I'd say please go ahead with your suggestion @vansterhant if you have time available to work with this!

Commit:
Refactored AttaibuteContentAreaRenderer

So that the standard renderer can perform the base duties
while delegating attribute related tasks to the contexts

Commit:
Added a documentation snippet to the readme

And bumped the version number
